api/models/schema.py

```diff
@@ -251,85 +252,47 @@ def validate_longitude(cls, v):
-        exclusiveMinimum= -180,
-        exclusiveMaximum= 180
-    )
-
-
-# class Line(GeoShape):
-#     line: str = Field(description="A line that is expressed as a series of two or more point objects separated by space.")
-
-#     @validator('line')
-#     def validate_line(cls, v):
-#         if not isinstance(v, str):
-#             raise TypeError('string required')
-#         v = v.strip()
-#         if not v:
-#             raise ValueError('empty string')
-#         v_parts = v.split(' ')
-#         if len(v_parts) < 2:
-#             raise ValueError('Line must have at least 2 points')
-#         for item in v_parts:
-#             try:
-#                 item = float(item)
-#             except ValueError:
-#                 raise ValueError('Line point is not a number')
-#             item = abs(item)
-#             if item > 180:
-#                 raise ValueError('Line point must be between -180 and 180')
-#         return v
-
-
-# class Polygon(GeoShape):
-#     polygon: str = Field(description="A polygon outlines the boundary of a specific region by a point-to-point path for which the starting and ending points are the same.")
-
-#     @validator('polygon')
-#     def validate_polygon(cls, v):
-#         if not isinstance(v, str):
-#             raise TypeError('string required')
-#         v = v.strip()
-#         if not v:
-#             raise ValueError('empty string')
-#         v_parts = v.split(' ')
-#         if len(v_parts) < 3:
-#             raise ValueError('Polygon must have at least 3 points')
-#         for item in v_parts:
-#             try:
-#                 item = float(item)
-#             except ValueError:
-#                 raise ValueError('Polygon point is not a number')
-#             item = abs(item)
-#             if item > 180:
-#                 raise ValueError('Polygon point must be between -180 and 180')
-#         return v
+    type: str = Field(
+        alias="@type",
+        default="GeoShape",
+        description="A structured representation that describes the coordinates of a geographic feature."
+    )
+    box: str = Field(
+        description="A box is a rectangular region defined by a pair of coordinates representing the "
+                    "southwest and northeast corners of the box."
+    )
+
+    @validator('box')
+    def validate_box(cls, v):
+        if not isinstance(v, str):
+            raise TypeError('string required')
+        v = v.strip()
+        if not v:
+            raise ValueError('empty string')
+        v_parts = v.split(' ')
+        if len(v_parts) != 4:
+            raise ValueError('Bounding box must have 4 coordinate points')
+        for index, item in enumerate(v_parts, start=1):
+            try:
+                item = float(item)
+            except ValueError:
+                raise ValueError('Bounding box coordinate is not a number')
+            item = abs(item)
```

Comment:
@pkdash Is this conversion to absolute value intended here given the conditionals below? `item < -180` will always evaluate to false.

Reply:
@Maurier Good catch. Will fix it.